docs/learn.md

```diff
@@ -1,3 +1,153 @@
 # Learn
 
-_TODO: This will be a one-page with "all the things" in it_
+## New to OCaml?
+
+As Melange is a fork of the OCaml compiler, this documentation is designed to
+include the differences between both. But for the parts that overlap between
+Melange and OCaml, we will always refer to the main OCaml documentation
+throughout these docs.
+
+If you are completely new to OCaml, you might want to get familiar first with
+the language, there are [plenty of resources available](https://ocaml.org/docs),
+but we recommend the following tutorials from the official OCaml website:
+
+- [A First Hour With OCaml](https://ocaml.org/docs/first-hour)
+- [OCaml Exercises](https://ocaml.org/problems)
+
+## Why Melange
+
+Melange is a fork of the OCaml compiler that produces JavaScript.
+
+OCaml offers an industrial-strength, state-of-the-art type system and provides
+type inference with very few type annotations, proving invaluable in managing
+large projects.
+
+JavaScript is one of the most pervasive platforms to deploy and run software.
+Thanks to years of efforts to improve the different VMs available, the
+JavaScript code running on browsers and other environments is heavily optimized
+and can support use cases for large products and tools.
+
+Melange helps developers and companies bring the advantages of the OCaml
+platform to users of the Web platform in a way that makes it easy for developers
+to integrate with both ecosystems.
+
+### A bit of history
+
+To better understand where Melange comes from, it might help to go through some
+of the related projects that have appeared over the last decade or so.
+
+#### Js_of_ocaml
+
+[Js_of_ocaml](https://github.com/ocsigen/js_of_ocaml/) is another OCaml to
+JavaScript compiler that was made public in 2011. In [the presentation
+paper](https://www.irif.fr/~balat/publications/vouillon_balat-js_of_ocaml.pdf)
+published in 2013, it's explicitly mentioned that one of its design goals was to
+remain as compatible as possible with the OCaml compiler, without requiring a
+lot of maintenance work, as the OCaml community was not excessively large at the
+time.
+
+To achieve this goal, Js_of_ocaml picks the bytecode generated by OCaml [batch
+compilation](https://ocaml.org/manual/comp.html) and generates JavaScript from
+it. OCaml bytecode has a very stable interface, so Js_of_ocaml can easily
+upgrade to new versions of the compiler. Due to this design decision, it can
+also remain compatible with most of the OCaml ecosystem, as long as the tools or
+libraries don't rely on C code.
+
+The downside of using bytecode is that it gets harder to communicate with
+existing JavaScript code. For example, a Js_of_ocaml string is not compiled as a
```

Comment:
Didn't they recently make strings compatible with the JS side?

Reply:
Right, I fixed this text in https://github.com/melange-re/documentation-site/commit/6cedebd48d2462bca74b4cee8d18952bd0d839d1.